### DISCUSSION TITLE
FeedType POST_ORDER_FULFILLMENT_DATA

### DISCUSSION THREAD

**Discussion by valle-gp** [abuzuhri/Amazon-SP-API-CSharp#339] [General]
Hi,

I am trying to update orders with shipping date but the feed always gives error: "The XML you submitted is ill-formed at the Amazon Envelope XML level at (or near) line 1, column 1"

This is the xml that che createdocument return
<?xml version="1.0" encoding="utf-16"?><AmazonEnvelope xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema"><Header><DocumentVersion>1.02</DocumentVersion><MerchantIdentifier>YYYYYYYYYYYYYYY</MerchantIdentifier></Header><MessageType>OrderFulfillment</MessageType><Message><MessageID>1</MessageID><OperationType>Update</OperationType><OrderFulfillment><AmazonOrderID>112-4291533-8809020</AmazonOrderID><FulfillmentDate>2022-09-28T16:00:17.317+02:00</FulfillmentDate><FulfillmentData><CarrierCode>DHL</CarrierCode><ShippingMethod>DHL Express</ShippingMethod><ShipperTrackingNumber>XXXXXXXXXXX</ShipperTrackingNumber></FulfillmentData></OrderFulfillment></Message></AmazonEnvelope>

How I can resolve it ?
Thanks

**Comment by valle-gp** [comment]
Hi Tareq ,
I found the problem; I was passing to amazonConnection.Feed.SubmitFeed the content saved in a file instead of directly the result of createDocument.GetXML()

Now, however, it gives error : We are unable to process the XML feed because one or more items are invalid. Please re-submit the feed.

In your example the item fields are not passed; what could be the problem?

**Comment by valle-gp** [comment]
Hi,

 I solve the problem; the date must be formatted in ISO8601 as well , so not 2020-06-23T04:25:09-0500 but 2020-06-22T11:25:09Z

**Reply by abuzuhri** [reply]
Good news , Please share the code that generate this XML to check why that

**Comment by valle-gp** [comment]
Hello,

this is the xml 
<AmazonEnvelope xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema">
  <Header>
    <DocumentVersion>1.02</DocumentVersion>
    <MerchantIdentifier>A3KJEBU6VS9RWN</MerchantIdentifier>
  </Header>
  <MessageType>OrderFulfillment</MessageType>
  <Message>
    <MessageID>1</MessageID>
    <OperationType>Update</OperationType>
    <OrderFulfillment>
      <AmazonOrderID>701-XXXXXXX-4706620</AmazonOrderID>
      <FulfillmentDate>2022-09-29T04:25:09-0500</FulfillmentDate>
      <FulfillmentData>
        <CarrierCode>UPS</CarrierCode>
        <ShippingMethod>UPS Express</ShippingMethod>
        <ShipperTrackingNumber>1ZW5426504945XXXXX</ShipperTrackingNumber>
      </FulfillmentData>
    </OrderFulfillment>
  </Message>
</AmazonEnvelope>
